noxfile.py

```diff
@@ -0,0 +1,70 @@
+import nox
+import os
+import sys
+
+
+@nox.session(name="pybamm-requires")
+def run_pybamm_requires(session):
+    if sys.platform != "win32" or sys.platform != "darwin":
+        session.install("wget", "cmake")
+        session.run("python", "scripts/install_KLU_Sundials.py")
+        session.run(
+            "git",
+            "clone",
+            "https://github.com/pybind/pybind11.git",
+            "pybind11/",
+            external=True,
+        )
+
+
+@nox.session(name="coverage")
+def run_coverage(session):
+    homedir = os.getenv("HOME")
+    session.env["SUNDIALS_INST"] = session.env.get("SUNDIALS_INST", f"{homedir}/.local")
+    session.env[
+        "LD_LIBRARY_PATH"
+    ] = f"{homedir}/.local/lib:{session.env.get('LD_LIBRARY_PATH')}"
+    session.install("coverage", "scikits.odes")
+    session.run("pip", "install", "-e", ".[dev]")
+    if sys.platform != "win32" or sys.platform != "darwin":
+        session.run("pybamm_install_jax")
```

Comment:
Can you refactor this step out as a new function? Maybe something like this -

```py
@nox.session(name="install-jax")
    def run_install_jax(session):
        if sys.platform != "win32":
             session.run("pybamm_install_jax")
```

and then execute it only once in the workflows?

Reply:
I can do this but I would still have to install `jax` explicitely in requiring sessions due to `isolated_build`.
That being said we are executing `pybamm_install_jax` two times (only on linux) which barely takes 10-15 seconds on CI.